src/analyze/analyze-verify.c

```diff
@@ -218,13 +231,20 @@ static int verify_unit(Unit *u, bool check_man) {
         return r;
 }
 
-int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run_generators) {
+static void clear_log_syntax_callback(void *dummyp) {
+          set_log_syntax_callback(/* cb= */ NULL, /* userdata= */ NULL);
+}
+
+int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run_generators, RecursiveErrors recursive_errors) {
         const ManagerTestRunFlags flags =
                 MANAGER_TEST_RUN_MINIMAL |
                 MANAGER_TEST_RUN_ENV_GENERATORS |
+                (recursive_errors == RECURSIVE_ERRORS_NO) * MANAGER_TEST_RUN_IGNORE_DEPENDENCIES |
                 run_generators * MANAGER_TEST_RUN_GENERATORS;
 
         _cleanup_(manager_freep) Manager *m = NULL;
+        _cleanup_set_free_free_ Set *s = NULL;
+        _unused_ _cleanup_(clear_log_syntax_callback) void *dummy;
```

Comment:
hmm, maybe we should actually use a type that is truly zero length here, because we don't really car about the variable. i.e. i wonder if an empty strict or zero-length array would be good here. or maybe a struct with a zero length array. i.e.

```c
typedef struct { 
        int _empty[0]; 
} dummy_t; 
```

at least on gcc that should compile fine and be a zero-length type.

Reply:
given it's in a "helper" command line tool that runs and exits, rather than a long-running service, is it really worth the additional complication to save a few bytes?

Reply:
well, it's nicer I think. Not because it saves memory (not sure if it actually does, maybe gcc is smart away to optimize this away anyway, dunno), but simply because we then have an explicit type for such "dummy" usecases, and that's what I like. 

I mean, the 5 lines I posted above actually work, I tested that here. Just copy that into macro.h or so, and then use dummy_t instead of void* here, it's a trivial addition and makes things a bit clearer I think when reading over it.

Reply:
I.e. I care about expressiveness, well readable, self-explanatory code. I think a `dummy_t` type could help there, since it makes very clear that the variable is a dummy, and not actually used as a variable.

Reply:
ok, makes sense

Reply:
So would the inline function clear_log_syntax_callback() in log.h then look like this:

static inline void clear_log_syntax_callback(dummy_t dummyp) {
          set_log_syntax_callback(/* cb= */ NULL, /* userdata= */ NULL);
}

Reply:
yes